client/js/store/modules/auth.js

```diff
@@ -1,8 +1,36 @@
-import { io } from '../../plugins/sails.js';
+import { io, reconnectSockets } from '../../plugins/sails.js';
 import { ROUTE_NAME_LOBBY, ROUTE_NAME_GAME } from '@/router';
 
 import { getPlayerPnumByUsername } from '_/utils/game-utils.js';
 
+async function handleLogin(context, username, password, signup = false) {
+  const type = signup ? 'signup' : 'login';
+  try {
+    const response = await fetch(`/user/${type}`, {
+      method: 'POST',
+      headers: new Headers({
+        'Content-Type': 'application/json',
+      }),
+      body: JSON.stringify({
+        username,
+        password,
+      }),
+      credentials: 'include',
+    });
+    const data = await response.json();
+    if (response.status !== 200) {
+      throw data.message;
+    }
+    // If the response was successful, the user is logged in
+    context.commit('authSuccess', username);
+    reconnectSockets();
+    return;
```

Comment:
Okay now I see. Socket would keep the old cookie value from its initial connection prior to auth

Reply:
Yes, that's exactly right. By disconnecting and reconnecting we force the socket to refresh the authentication via handshake. There's probably a better way but I'm not aware of it. I know that sockets and REST calls share middleware in sails, but the reality is that the sockets don't set cookies on the browser which was causing problems.